src/to-be-in-the-document.ts

```diff
@@ -0,0 +1,39 @@
+import type { ReactTestInstance } from 'react-test-renderer';
+import { screen } from '@testing-library/react-native';
+import { matcherHint, RECEIVED_COLOR } from 'jest-matcher-utils';
+import { checkReactElement, printElement } from './utils';
+
+export function toBeInTheDocument(this: jest.MatcherContext, element: ReactTestInstance) {
+  if (element !== null) {
+    checkReactElement(element, toBeInTheDocument, this);
+  }
+
+  const pass = element === null ? false : screen.container === getRootElement(element);
+
+  const errorFound = () => {
+    return `expected document not to contain element but found:\n${printElement(element)}`;
+  };
+
+  const errorNotFound = () => {
+    return `element could not be found in the document`;
```

Comment:
A good error message has a recovery hint. Something like "Please make sure you rendered the element." would be welcome